homeassistant/components/mqtt/client.py

```diff
@@ -380,7 +376,7 @@ def __init__(
         self._simple_subscriptions: defaultdict[str, set[Subscription]] = defaultdict(
             set
         )
-        self._wildcard_subscriptions: set[Subscription] = set()
+        self._wildcard_subscriptions: dict[Subscription, None] = {}
```

Comment:
Let's add a comment to explain we do this to basically make `self._wildcard_subscriptions` an ordered set

Reply:
Done